src/test/ui/associated-types/hr-associated-type-bound-2.rs

```diff
@@ -8,14 +8,13 @@ where
     }
 }
 
-impl X<'_> for u32
+impl X<'_> for u32 //~ overflow evaluating the requirement `for<'b> u32: X<'b>`
```

Comment:
This is not ideal. This is probably a blocker for merging this PR.

Reply:
Actually, this is what we get in the more general case *without* a bound var: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f938d8e6a91f2162cf10896d2b5bafb5

I don't think this is blocking (though I think we *can* actually handle this correctly; I don't think that's in scope for this PR though)